api/src/Auth/Driver/Guard.php

```diff
@@ -0,0 +1,351 @@
+    {
+        if ($this->user === false){
+            return null;
+        }
+
+        if ($this->user){
+            return $this->user;
+        }
+
+        try{
+            $user = $this->getUserFromRequest();
+        }catch (\Exception $e){
+            return null;
+        }
+
+        if ($user){
+            $this->user = $user;
+            $this->viaToken = true;
+        }
+
+        return $user;
+    }
+
+    public function getUserFromRequest()
+    {
+        $token = $this->getRequestParser()->getToken($this->getRequest());
+        $payload = $this->getTokenizer()->decode($token);
+        $this->getValidationFactory()->validatePayload($payload);
+        $user = $this->getProvider()->retrieveByToken(null, $payload);
+        return $user;
+    }
+
+    /**
+     * Log a user into the application without token.
+     *
+     * @param  array $credentials
+     * @return bool
+     */
+    public function once(array $credentials = [])
+    {
+        return $this->attempt($credentials, false, true);
+    }
+
+    /**
+     * Attempt to authenticate a user using the given credentials.
+     *
+     * @param  array $credentials
+     * @param  bool $remember
+     * @param  bool $login
+     * @return bool
+     */
+    public function attempt(array $credentials = [], $remember = true, $login = true)
+    {
+        $user = $this->getProvider()->retrieveByCredentials($credentials);
+        if ($user  && $this->getProvider()->validateCredentials($user, $credentials)){
+            if ($login) {
+                $this->login($user, $remember);
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Attempt to authenticate using HTTP Basic Auth.
+     *
+     * @param  string $field
+     * @return null|\Symfony\Component\HttpFoundation\Response
+     * @throws \Exception
+     */
+    public function basic($field = 'email')
+    {
+        throw new \Exception('Not implemented');
+    }
+
+    /**
+     * Perform a stateless HTTP Basic login attempt.
+     *
+     * @param  string $field
+     * @return null|\Symfony\Component\HttpFoundation\Response
+     * @throws \Exception
+     */
+    public function onceBasic($field = 'email')
+    {
+        throw new \Exception('Not implemented');
+    }
+
+    /**
+     * Validate a user's credentials.
+     *
+     * @param  array $credentials
+     * @return bool
+     * @throws \Exception
+     */
+    public function validate(array $credentials = [])
+    {
+        throw new \Exception('Not implemented');
+    }
+
+    /**
+     * Log a user into the application.
+     *
+     * @param  \Illuminate\Contracts\Auth\Authenticatable $user
+     * @param  bool $remember set or update token
+     * @return void
+     */
+    public function login(Authenticatable $user, $remember = false)
+    {
+        $this->user = $user;
+        if ($remember){
+            $user->setRememberToken($this->generateToken($user));
+        }
```

Comment:
I'm not sure if you're conforming to a service here, but `$remember` makes no sense in the context of jwt stateless auth as no tokens are persisted anywhere - it is on the client to remember or not.

I get the feeling that using the `$user->setRememberToken` method is confusing the intent of jwt.